pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -171,6 +200,97 @@ func (p *MellanoxPlugin) OnNodeStateChange(old, new *sriovnetworkv1.SriovNetwork
 	return
 }
 
+func (p *MellanoxPlugin) updateNicSpec(state *sriovnetworkv1.SriovNetworkNodeState) {
+	p.nicSyncList.Lock()
+	defer p.nicSyncList.Unlock()
+
+	var interfaces sriovnetworkv1.Interfaces
+OUTER:
+	for _, ifaceStatus := range state.Status.Interfaces {
+		if ifaceStatus.Vendor != MellanoxVendorId {
+			continue
+		}
+		for _, ifaceSpec := range state.Spec.Interfaces {
+			if ifaceSpec.PciAddress != ifaceStatus.PciAddress {
+				continue
+			}
+			// Skip non RDMA NICs
+			if !ifaceSpec.IsRdma {
+				continue OUTER
+			}
+			if ifaceSpec.NumVfs == 0 {
+				continue OUTER
+			}
+
+			interfaces = append(interfaces, ifaceSpec)
```

Comment:
why are you not using ifaceStatus ? is it because OnNodeStateChange is not called on `Status` field change

Reply:
IfaceSpec determines if this NIC has a policy affecting it which is not clear by IfaceStatus, which make it better choice

Reply:
would appending `ifaceStatus` instead of `ifaceSpec` work the same ? (thus avoiding method param change in utils.go)